monero-sys/src/lib.rs

```diff
@@ -470,6 +495,39 @@ impl WalletHandle {
         self.call(move |wallet| wallet.main_address()).await
     }
 
+    /// Compute subaddress summaries for an account on the wallet thread.
+    pub async fn subaddress_summaries(
+        &self,
+        account_index: u32,
+    ) -> anyhow::Result<Vec<SubaddressSummary>> {
+        self.call(move |wallet| wallet.subaddress_summaries_sync(account_index))
+            .await
+    }
+
+    /// Create a new subaddress in the specified account and persist changes.
+    pub async fn create_subaddress(&self, account_index: u32, label: String) -> anyhow::Result<()> {
+        self.call(move |wallet| wallet.add_subaddress(account_index, &label))
+            .await
+            .context("Failed to add subaddress")?;
+        // Persist changes on disk
+        self.store_in_current_file().await?;
```

Comment:
This makes any call to this function very expensive. Saving can take a long while sometimes because it'll wait for a bunch of Mutexes of wallet2 to become unlocked.

I think we should just rely on the periodic wallet save / save on close.